desc/geometry/surface.py

```diff
@@ -357,6 +361,200 @@ def from_near_axis(cls, aspect_ratio, elongation, mirror_ratio, axis_Z, NFP=1):
+        full_output : bool, optional
+            If True, also return a dict of useful data about the surfaces and a
+            tuple where the first element is the residual from
+            the root finding and the second is the number of iterations.
+
+        Returns
+        -------
+        offset_surface : FourierRZToroidalSurface
+            FourierRZToroidalSurface, created from fitting points offset from the input
+            surface by the given constant offset.
+        data : dict
+            dictionary containing  the following data, in the cylindrical basis:
+                ``n`` : (``grid.num_nodes`` x 3) array of the unit surface normal on
+                    the base_surface evaluated at the input ``grid``
+                ``x`` : (``grid.num_nodes`` x 3) array of the position vectors on
+                    the base_surface evaluated at the input ``grid``
+                ``x_offset_surface`` : (``grid.num_nodes`` x 3) array of the
+                    position vectors on the offset surface, corresponding to the
+                    ``x`` points on the base_surface (i.e. the points to which the
+                    offset surface was fit)
+        info : tuple
+            2 element tuple containing residuals and number of iterations
+            for each point. Only returned if ``full_output`` is True
+
+        """
+        if grid is None:
+            grid = LinearGrid(
+                M=base_surface.M,
+                N=base_surface.N,
+                NFP=base_surface.NFP,
+                sym=base_surface.sym,
+            )
+        assert isinstance(
+            base_surface, FourierRZToroidalSurface
+        ), "base_surface must be a FourierRZToroidalSurface!"
+        M = base_surface.M if M is None else int(M)
+        N = base_surface.N if N is None else int(N)
+
+        Rbasis = base_surface.R_basis
+        Zbasis = base_surface.Z_basis
+
+        def n_and_r_jax(nodes):
+            R = Rbasis.evaluate(nodes) @ base_surface.R_lmn
+            Z = Zbasis.evaluate(nodes) @ base_surface.Z_lmn
+            R_t = Rbasis.evaluate(nodes, np.array([0, 1, 0])) @ base_surface.R_lmn
+            Z_t = Zbasis.evaluate(nodes, np.array([0, 1, 0])) @ base_surface.Z_lmn
+            R_z = Rbasis.evaluate(nodes, np.array([0, 0, 1])) @ base_surface.R_lmn
+            Z_z = Zbasis.evaluate(nodes, np.array([0, 0, 1])) @ base_surface.Z_lmn
+
+            phi = nodes[:, 2]
+            coords = jnp.stack([R, phi, Z], axis=1)
+            re = rpz2xyz(coords)
+            # NOTE: when generalized toroidal angle is implemented
+            # this must include omega
+            e_theta = jnp.array([R_t, jnp.zeros_like(R_t), Z_t]).T
+            e_zeta = jnp.array([R_z, R, Z_z]).T
+            e_theta_cross_e_zeta = cross(e_theta, e_zeta)
+            n = safediv(
+                e_theta_cross_e_zeta.T, jnp.linalg.norm(e_theta_cross_e_zeta, axis=1)
+            ).T
+            n = rpz2xyz_vec(n, phi=phi)
```

Comment:
Why not use the usual compute interface for these quantities, like `base_surface.compute("|e_theta x e_zeta|", grid=Grid(nodes))`?

Reply:
we can't `jit` it if we go through the compute interface as the creation of Transform objects is not jittable

Reply:
It is if you make method="jitable"

Reply:
That works and I made the change, needed a bug fix

I will say that the previous method is about 10-20% faster than going through the compute function, but that is not a huge change.